### PR TITLE
[CCIP-2498] Handling Unregister*PluginLpFilters for Exec + Commit in a LOOP compatible way

### PR DESCRIPTION
**Pull request by patrickhuie19**
## Motivation
When jobs are deleted, `UnregisterExecPluginLpFilters` is called. It currently has EVM specific dependencies. 

## Solution
Rely on the `Close` method in the src and dst provider to unregister the necessary lp filters. 

## Follow-on Work
More comprehensive job deletion unit testing for CCIP exec + commit jobs: CCIP-2847

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/ccipexec/initializers_test.go [hunk 1]
```diff
@@ -1,16 +1,108 @@
...
+	//TODO implement me
+	panic("implement me")
+}
+
+func (m MockCCIPExecProvider) SourceNativeToken(ctx context.Context, addr cciptypes.Address) (cciptypes.Address, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
 func TestGetExecutionPluginFilterNamesFromSpec(t *testing.T) {
```
Comment (patrickhuie19) [inline core/services/ocr2/plugins/ccip/ccipexec/initializers_test.go]: This test is now effectively meaningless - I'll spend some thinking about what a useful unit test would look like here.

File: core/services/ocr2/plugins/ccip/ccipexec/initializers.go [hunk 2]
```diff
@@ -199,33 +199,17 @@ func NewExecServices(ctx context.Context, lggr logger.Logger, jb job.Job, srcPro
...
-		return err
-	}
-
-	offRampAddress, err := params.offRampReader.Address(ctx)
-	if err != nil {
-		return fmt.Errorf("get offramp reader address: %w", err)
-	}
-
-	versionFinder := factory.NewEvmVersionFinder()
+// This currently works because the filters registered by the created custom providers when the job is first added
```
Comment (patrickhuie19) [inline core/services/ocr2/plugins/ccip/ccipexec/initializers.go:199]: @dimkouv these assumptions are probably most important to review on this PR.

File: core/services/relay/evm/commit_provider.go [hunk 3]
```diff
@@ -84,149 +95,198 @@ func NewDstCommitProvider(
...
 }
 
-func (P DstCommitProvider) Name() string {
+func (P *DstCommitProvider) Name() string {
 	return "CCIPCommitProvider.DstRelayerProvider"
 }
 
-func (P DstCommitProvider) Close() error {
-	return nil
+func (P *DstCommitProvider) Close() error {
```
Comment (dimkouv) [inline core/services/relay/evm/commit_provider.go:165]: (not related to the PR) wondering why Receiver is a capital `P` :S

File: core/services/relay/evm/commit_provider.go [hunk 4]
```diff
@@ -84,149 +95,198 @@ func NewDstCommitProvider(
...
+	versionFinder := ccip.NewEvmVersionFinder()
+
+	unregisterFuncs := make([]func() error, 0, 2)
```
Comment (dimkouv) [inline core/services/relay/evm/commit_provider.go:168]: could initialize the slice directly without appending twice

File: core/services/relay/evm/exec_provider.go [hunk 5]
```diff
@@ -221,79 +258,105 @@ func NewDstExecProvider(
...
-func (d DstExecProvider) Close() error {
-	return nil
+// Close is called when the job that created this provider is deleted
+// At this time, any of the methods on the provider may or may not have been called.
+// If NewOnRampReader and NewCommitStoreReader have not been called, their corresponding
+// Close methods will be expected to error.
+func (d *DstExecProvider) Close() error {
+	versionFinder := ccip.NewEvmVersionFinder()
+
+	unregisterFuncs := make([]func() error, 0, 2)
```
Comment (dimkouv) [inline core/services/relay/evm/exec_provider.go:284]: same, no need to append

**Review by dimkouv — APPROVED** [review]
Looks like a very good change. Much cleaner management of the filters.
We need to trigger some e2e tests that are updating jobSpecs, triggering job restarts/ etc.. to make sure everything's working as expected. But overall LGTM, just minor comments.

**Review by krehermann — APPROVED** [review]

File: core/services/ocr2/delegate.go [hunk 6]
```diff
@@ -340,15 +341,48 @@ func (d *Delegate) cleanupEVM(ctx context.Context, jb job.Job, relayID types.Rel
...
+		if err2 != nil {
+			return err2
+		}
+
+		srcProvider, _, err2 := d.ccipCommitGetSrcProvider(ctx, jb, pluginJobSpecConfig, transmitterID, dstProvider)
+		if err2 != nil {
+			return err2
+		}
+		err2 = ccipcommit.UnregisterCommitPluginLpFilters(srcProvider, dstProvider)
 		if err != nil {
```
Comment (0xnogo) [inline core/services/ocr2/delegate.go:361]: Do you mean `if err2 != nil`?

**Pull request merged**